### ISSUE TITLE
Documentation: `num_dims` must be set on auto-embedding field for vector queries to work

### ISSUE THREAD

**Issue by juszzz** [typesense/typesense#1455]
## Description

Generating auto-embeddings by following the steps documented [here](https://typesense.org/docs/0.25.1/api/vector-search.html#option-b-auto-embedding-generation-within-typesense), _without_ including the `num_dims` configuration parameter appropriate for the auto-embedding model in question will result in a 400-class error and a `Field 'XYZ' does not have a vector query index` message. I believe this is a simple documentation oversight, and that adding `num_dims` to the config blocks for the various auto-embedding examples will help other users avoid this somewhat cryptic error message.

## Steps to reproduce

Follow the [instructions](https://typesense.org/docs/0.25.1/api/vector-search.html#option-b-auto-embedding-generation-within-typesens) to generate an auto-embedding vector field.

## Expected Behavior

Running a [vector search query](https://typesense.org/docs/0.25.1/api/vector-search.html#querying-for-similar-documents) results in retrieved results.

## Actual Behavior

The server returns a 400-class error with the message `Field 'XYZ' does not have a vector query index`. This is due to the fact that the documentation for auto-embeddings does not indicate that `num_dims` is a required additional configuration parameter and must be set with a value appropriate to the embeddings model being used (e.g., `768` for `gcp/textembedding-gecko@latest`, i.e., Vertex API).

## Metadata

**Typesense Version**: 0.25.0

**OS**: N/A

**Comment by juszzz** [comment]
Related issue that helped me figure out what I was missing in my configuration: #877.

**Comment by jasonbosco** [comment]
@juszzz Auto-embedding does not require you to specify a `num_dim` property for the field, since it is automatically derived from the model automatically. 

I suspect what happened is that the auto-embedding code snippets use the vector field name as `embedding` as an example, but the search code snippets use the vector field as `vec` as an example. Could you switch `vec` to `embedding` in the search code snippet and see if you still get the error?

If you do, could you share with me the exact curl commands you're running verbatim (minus the API key)?

**Comment by juszzz** [comment]
Thanks for the quick reply @jasonbosco , as usual! I actually tried making search commands with both `vec`, and `embedding`, and neither worked for me. I even switched the indexed field name to my own custom name, `productEmbedding`, and tried querying that is well, with the same result. Only when I added a `num_dims` field to the Vertex AI embedding configuration while indexing was I able to get search to work.

**Comment by jasonbosco** [comment]
Hmm, could you try replicating this in `0.25.2.rc15` as well? 

Could you also share with me the exact curl commands you're running verbatim (minus the API key)?

**Comment by juszzz** [comment]
Unfortunately I don't control our TS deployment, so I won't be able to update the server version (and the person who does is out until the new year).

I'm not using curl, but rather the Python API. Here's the relevant configuration. Schema:
```
base_products_schema = {
    "name": "foo",
    "enable_nested_fields": True,
    "fields": [
        {"name": "id", "type": "string"},
        {"name": "name", "type": "string", "sort": True},
        {"name": "categories", "type": "string[]"},
        {"name": "specificationArray", "type": "string[]"},
        {
            "name" : "productEmbedding",
            "type" : "float[]",
            "num_dim": 768,
            "index": True,
            "embed": {
                "from": [
                    "name",
                    "categories",
                    "specificationArray"
                ],
                "model_config": {
                    "model_name": "gcp/textembedding-gecko@latest",
                    "access_token": "***********",
                    "refresh_token": "***********",
                    "client_id": "************", 
                    "client_secret": "**********",
                    "project_id": "*************"
                }
            }
        }
    ],
}
```

Search query:
```
client.collections["foo"].documents.search({'q': '*', 'vector_query': 'productEmbedding:([], id: foo)'})
```.

When I omit `num_dims` from the schema, the indexing succeeds but the query fails with the error message described above. When I include it, the query succeeds.

**Comment by ozanarmagan** [comment]
@juszzz I couldn't reproduce your problem with `0.25.2rc14`, so updating Typesense should fix your problem. Can you confirm this by running Typesense on your local with Docker?

**Comment by juszzz** [comment]
@ozanarmagan thanks for the pointer. It seems we were using `0.25.0` -- when I use that Docker image locally, I'm able to reproduce the error. When I switch to `0.25.1` (or `0.25.2.rc14`), the error goes away. So I think the issue has been resolved as of v0.25.1, which is great. Maybe the only small thing to address here is that [the documentation for v0.25.0](https://typesense.org/docs/0.25.0/api/vector-search.html#using-gcp-vertex-ai-api) does not include `num_dims` as a required configuration parameter, which it seems that it needs to be. Thanks again for the assistance!
